Add new plugin list to plugin_introduction.adoc

FlowWriter opened the pull request:
Remember you reviewed this because it's our June blog post. I just removed the blog-related info and formatted it toward the end of the chapter/file.

Commit:
Update plugin_introduction.adoc

pierre commented on userguide/tutorials/plugin_introduction.adoc:
```diff
@@ -62,6 +62,137 @@ Kill Bill Plugins have full power (and therefore need to be designed and tested
 * They receive all events in the system.
 * They are isolated from the rest of the code and can use their own libraries (versions) without risk of conflict.
 
+== List of Open-Source Plugins
```
Should we merge this with the `Examples of Plugins` section above?

FlowWriter replied:
@pierre - Yes, I needed to do more work on this, didn't I? ;-)

Commit:
merge two sections

Moved info from "Examples of Plugins" into new plugins list. Moved other sections around for better flow of information. Renamed a section to "Plugin Capabilities."
